### PR TITLE
validate type of casted value

### PR DESCRIPTION
**Pull request by philebert**
validation fixed, see https://github.com/p1c2u/openapi-core/issues/105

### CODE REVIEW

**Comment by mnboos** [comment]
Is there a chance this will be merged soon?

**Comment by andyceo** [comment]
yeah, would be nice to have this fixed! @p1c2u sorry for troubling you :)

File: tests/integration/test_petstore.py [hunk 1]
```diff
@@ -238,8 +238,8 @@ def test_get_pets_ids_param(self, spec, response_validator):
         host_url = 'http://petstore.swagger.io/v1'
         path_pattern = '/v1/pets'
         query_params = {
-            'limit': '20',
-            'ids': ['12', '13'],
+            'limit': 20,
```
Comment (p1c2u) [inline tests/integration/test_petstore.py]: Query parameters are always passed as strings.

File: tests/integration/test_petstore.py [hunk 2]
```diff
@@ -782,10 +782,10 @@ def test_post_pets_raises_invalid_server_error(self, spec):
         }
         data = json.dumps(data_json)
         headers = {
-            'api_key': '12345',
+            'api_key': 12345,
```
Comment (p1c2u) [inline tests/integration/test_petstore.py]: Headers are always passed as strings.

File: tests/integration/test_petstore.py [hunk 3]
```diff
@@ -782,10 +782,10 @@ def test_post_pets_raises_invalid_server_error(self, spec):
...
         }
         cookies = {
-            'user': '123',
+            'user': 123,
```
Comment (p1c2u) [inline tests/integration/test_petstore.py]: Cookies are always passed as strings.

**Comment by p1c2u** [comment]
Closing in favor of #123

**Pull request closed without merging**